File: cli/internal/agents/agents.go
// Package agents lists every AI tool dot-folder we know about and the human
// display name we use for it in the bootstrap multi-select. This is the
// canonical list for the project — the Python side no longer carries one
// (the legacy `gather` command was its only consumer).
package agents

import "sort"

// Target is one row in the agent multi-select.
type Target struct {
	DotDir    string // e.g. ".claude" (relative under $HOME or .)
	Display   string // shown in the TUI, e.g. "Claude Code"
	Universal bool   // true if selected by default and can't be toggled off
	UnderHome bool   // true if the install path lives under $HOME (vs cwd)
}

// SkillsDir returns the absolute SKILL.md folder for this target.
// The skills-registry SKILL.md goes at SkillsDir(home, cwd) + "/skills-registry".
func (t Target) SkillsDir(home, cwd string) string {
	if t.UnderHome {
		return home + "/" + t.DotDir + "/skills"
	}
	return cwd + "/" + t.DotDir + "/skills"
}

// All returns every known agent target, sorted by display name. The
// `.agents` project-local folder is marked Universal so it's always
// selected in the multi-select.
func All() []Target {
	out := make([]Target, 0, len(known))
	out = append(out, known...)
	sort.SliceStable(out, func(i, j int) bool {
		if out[i].Universal != out[j].Universal {
			return out[i].Universal
		}
		return out[i].Display < out[j].Display
	})
	return out
}

// known is the curated list. Display names sourced from each tool's
// documentation / brand guidelines.
var known = []Target{
	// Universal (project-local, picked up by most agents)
	{DotDir: ".agents", Display: "Universal (.agents/skills)", Universal: true, UnderHome: false},

	// Home-directory agents
	{DotDir: ".claude", Display: "Claude Code", UnderHome: true},
	{DotDir: ".claude-code", Display: "Claude Code (legacy)", UnderHome: true},
	{DotDir: ".factory", Display: "Factory", UnderHome: true},
	{DotDir: ".codex", Display: "Codex CLI", UnderHome: true},
	{DotDir: ".cursor", Display: "Cursor", UnderHome: true},
	{DotDir: ".junie", Display: "Junie", UnderHome: true},
	{DotDir: ".aider", Display: "Aider", UnderHome: true},
	{DotDir: ".continue", Display: "Continue", UnderHome: true},
	{DotDir: ".windsurf", Display: "Windsurf", UnderHome: true},
	{DotDir: ".codeium", Display: "Codeium", UnderHome: true},
	{DotDir: ".zed", Display: "Zed", UnderHome: true},
	{DotDir: ".anthropic", Display: "Anthropic", UnderHome: true},
	{DotDir: ".openai", Display: "OpenAI", UnderHome: true},
	{DotDir: ".cline", Display: "Cline", UnderHome: true},
	{DotDir: ".roo", Display: "Roo", UnderHome: true},
	{DotDir: ".roocode", Display: "Roo Code", UnderHome: true},
	{DotDir: ".gemini", Display: "Gemini", UnderHome: true},
	{DotDir: ".antigravity", Display: "Antigravity", UnderHome: true},
	{DotDir: ".aider-desk", Display: "Aider Desk", UnderHome: true},
	{DotDir: ".augment", Display: "Augment", UnderHome: true},
	{DotDir: ".bob", Display: "Bob", UnderHome: true},
	{DotDir: ".codeartsdoer", Display: "CodeArts Doer", UnderHome: true},
	{DotDir: ".codebuddy", Display: "CodeBuddy", UnderHome: true},
	{DotDir: ".codemaker", Display: "CodeMaker", UnderHome: true},
	{DotDir: ".codestudio", Display: "Code Studio", UnderHome: true},
	{DotDir: ".commandcode", Display: "Command Code", UnderHome: true},
	{DotDir: ".copilot", Display: "GitHub Copilot", UnderHome: true},
	{DotDir: ".cortex", Display: "Cortex", UnderHome: true},
	{DotDir: ".crush", Display: "Crush", UnderHome: true},
	{DotDir: ".deepagents", Display: "DeepAgents", UnderHome: true},
	{DotDir: ".devin", Display: "Devin", UnderHome: true},
	{DotDir: ".firebender", Display: "Firebender", UnderHome: true},
	{DotDir: ".forge", Display: "Forge", UnderHome: true},
	{DotDir: ".goose", Display: "Goose", UnderHome: true},
	{DotDir: ".iflow", Display: "iFlow", UnderHome: true},
	{DotDir: ".kilocode", Display: "Kilo Code", UnderHome: true},
	{DotDir: ".kiro", Display: "Kiro", UnderHome: true},
	{DotDir: ".kode", Display: "Kode", UnderHome: true},
	{DotDir: ".mcpjam", Display: "MCPJam", UnderHome: true},
	{DotDir: ".mux", Display: "Mux", UnderHome: true},
	{DotDir: ".opencode", Display: "OpenCode", UnderHome: true},
	{DotDir: ".openhands", Display: "OpenHands", UnderHome: true},
	{DotDir: ".pi", Display: "Pi", UnderHome: true},
	{DotDir: ".qoder", Display: "Qoder", UnderHome: true},
	{DotDir: ".qwen", Display: "Qwen Code", UnderHome: true},
	{DotDir: ".rovodev", Display: "Rovo Dev", UnderHome: true},
	{DotDir: ".tabnine", Display: "Tabnine", UnderHome: true},
	{DotDir: ".trae", Display: "Trae", UnderHome: true},
	{DotDir: ".trae-cn", Display: "Trae CN", UnderHome: true},
	{DotDir: ".vibe", Display: "Vibe", UnderHome: true},
	{DotDir: ".zencoder", Display: "Zencoder", UnderHome: true},
	{DotDir: ".neovate", Display: "Neovate", UnderHome: true},
	{DotDir: ".pochi", Display: "Pochi", UnderHome: true},
	{DotDir: ".adal", Display: "Adal", UnderHome: true},
	{DotDir: ".snowflake", Display: "Snowflake", UnderHome: true},
}

File: mac-app/Sources/SkillsRegistryCore/Agents.swift
import Foundation

/// One known AI-tool dot-folder. Port of `cli/internal/agents/agents.go`.
public struct AgentTarget: Hashable, Sendable {
    public var dotDir: String   // e.g. ".claude"
    public var display: String  // e.g. "Claude Code"
    public var universal: Bool  // project-local, always relevant
    public var underHome: Bool  // install path under $HOME vs cwd

    public init(dotDir: String, display: String, universal: Bool = false, underHome: Bool = true) {
        self.dotDir = dotDir
        self.display = display
        self.universal = universal
        self.underHome = underHome
    }

    /// Absolute `<dot>/skills` directory for this target.
    public func skillsDir(home: String, cwd: String) -> String {
        let base = underHome ? home : cwd
        return base + "/" + dotDir + "/skills"
    }
}

public enum Agents {
    /// Every known target, sorted (universal first, then by display name).
    public static func all() -> [AgentTarget] {
        known.sorted { a, b in
            if a.universal != b.universal { return a.universal }
            return a.display < b.display
        }
    }

    /// Just the dot-dir names — used to enumerate candidate source folders.
    public static func dotDirs() -> [String] { known.map(\.dotDir) }

    static let known: [AgentTarget] = [
        AgentTarget(dotDir: ".agents", display: "Universal (.agents/skills)", universal: true, underHome: false),
        AgentTarget(dotDir: ".claude", display: "Claude Code"),
        AgentTarget(dotDir: ".claude-code", display: "Claude Code (legacy)"),
        AgentTarget(dotDir: ".factory", display: "Factory"),
        AgentTarget(dotDir: ".codex", display: "Codex CLI"),
        AgentTarget(dotDir: ".cursor", display: "Cursor"),
        AgentTarget(dotDir: ".junie", display: "Junie"),
        AgentTarget(dotDir: ".aider", display: "Aider"),
        AgentTarget(dotDir: ".continue", display: "Continue"),
        AgentTarget(dotDir: ".windsurf", display: "Windsurf"),
        AgentTarget(dotDir: ".codeium", display: "Codeium"),
        AgentTarget(dotDir: ".zed", display: "Zed"),
        AgentTarget(dotDir: ".anthropic", display: "Anthropic"),
        AgentTarget(dotDir: ".openai", display: "OpenAI"),
        AgentTarget(dotDir: ".cline", display: "Cline"),
        AgentTarget(dotDir: ".roo", display: "Roo"),
        AgentTarget(dotDir: ".roocode", display: "Roo Code"),
        AgentTarget(dotDir: ".gemini", display: "Gemini"),
        AgentTarget(dotDir: ".antigravity", display: "Antigravity"),
        AgentTarget(dotDir: ".aider-desk", display: "Aider Desk"),
        AgentTarget(dotDir: ".augment", display: "Augment"),
        AgentTarget(dotDir: ".bob", display: "Bob"),
        AgentTarget(dotDir: ".codeartsdoer", display: "CodeArts Doer"),
        AgentTarget(dotDir: ".codebuddy", display: "CodeBuddy"),
        AgentTarget(dotDir: ".codemaker", display: "CodeMaker"),
        AgentTarget(dotDir: ".codestudio", display: "Code Studio"),
        AgentTarget(dotDir: ".commandcode", display: "Command Code"),
        AgentTarget(dotDir: ".copilot", display: "GitHub Copilot"),
        AgentTarget(dotDir: ".cortex", display: "Cortex"),
        AgentTarget(dotDir: ".crush", display: "Crush"),
        AgentTarget(dotDir: ".deepagents", display: "DeepAgents"),
        AgentTarget(dotDir: ".devin", display: "Devin"),
        AgentTarget(dotDir: ".firebender", display: "Firebender"),
        AgentTarget(dotDir: ".forge", display: "Forge"),
        AgentTarget(dotDir: ".goose", display: "Goose"),
        AgentTarget(dotDir: ".iflow", display: "iFlow"),
        AgentTarget(dotDir: ".kilocode", display: "Kilo Code"),
        AgentTarget(dotDir: ".kiro", display: "Kiro"),
        AgentTarget(dotDir: ".kode", display: "Kode"),
        AgentTarget(dotDir: ".mcpjam", display: "MCPJam"),
        AgentTarget(dotDir: ".mux", display: "Mux"),
        AgentTarget(dotDir: ".opencode", display: "OpenCode"),
        AgentTarget(dotDir: ".openhands", display: "OpenHands"),
        AgentTarget(dotDir: ".pi", display: "Pi"),
        AgentTarget(dotDir: ".qoder", display: "Qoder"),
        AgentTarget(dotDir: ".qwen", display: "Qwen Code"),
        AgentTarget(dotDir: ".rovodev", display: "Rovo Dev"),
        AgentTarget(dotDir: ".tabnine", display: "Tabnine"),
        AgentTarget(dotDir: ".trae", display: "Trae"),
        AgentTarget(dotDir: ".trae-cn", display: "Trae CN"),
        AgentTarget(dotDir: ".vibe", display: "Vibe"),
        AgentTarget(dotDir: ".zencoder", display: "Zencoder"),
        AgentTarget(dotDir: ".neovate", display: "Neovate"),
        AgentTarget(dotDir: ".pochi", display: "Pochi"),
        AgentTarget(dotDir: ".adal", display: "Adal"),
        AgentTarget(dotDir: ".snowflake", display: "Snowflake"),
    ]
}
